lib/crypto/ed25519/ed25519.go

```diff
@@ -44,6 +44,22 @@ type PublicKey ed25519.PublicKey
 // PublicKeyBytes is an encoded ed25519 public key
 type PublicKeyBytes [PublicKeyLength]byte
 
+// VerifySignature verifies a signature given a public key and a message
+func VerifySignature(publicKey, signature, message []byte) error {
+	pubKey, err := NewPublicKey(publicKey)
+	if err != nil {
+		return fmt.Errorf("ed25519: %w", err)
+	}
+
+	_, err = pubKey.Verify(message, signature)
```

Comment:
you need to check the `bool` returned here, as the verification might fail, but there might not be an error

Reply:
its okay to do something like this:

```go
ok, err := pubKey.Verify(message, signature)
    if err != nil || !ok {
        return fmt.Errorf("ed25519: %w: for message 0x%x, signature 0x%x and public key 0x%x",
	    crypto.ErrSignatureVerificationFailed, message, signature, publicKey)
    }
```

since we don't return a bool anymore?

Reply:
Just have a single error check `err != nil` I guess, that should work.

Reply:
the `err != nil` verification already existed, what I did understand was that here we need to check the bool `i.e ok`, 'cause it can fail and `err` be `nil`

Reply:
done in da14c99